src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2077,19 +2077,112 @@ public async Task SendHttpGetRequest_WithCachePolicyBypassCache_DoNotAddHeaders(
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        var data = await connection.ReadRequestDataAsync();
+                        Assert.Equal(text, Encoding.UTF8.GetString(data.Body[0..text.Length]));
+                    });
+                }
+            );
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(0.1)]
+        public async Task SendHttpPost_WriteBufferingEnabled(int delaySec)
+        {
+            const string text = "Hello World!!!!\n";
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    const int size = 1048576 / 4;
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.AllowWriteStreamBuffering = true;
+                    request.ContentType = "application/text";
+                    request.ContentLength = size;
+                    request.Method = "POST";
+                    request.Timeout = (size/16 * delaySec) + request.Timeout;
+                    using var stream = await request.GetRequestStreamAsync();
+                    for (int i = 0; i < size/16; i++)
+                    {
+                        await stream.WriteAsync(new ReadOnlyMemory<byte>(Encoding.UTF8.GetBytes(text)));
+                        if (delaySec > 0)
+                        {
+                            await Task.Delay(TimeSpan.FromSeconds(delaySec));
+                        }
+                    }
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        var data = await connection.ReadRequestDataAsync();
+                        Assert.Equal(text, Encoding.UTF8.GetString(data.Body[0..text.Length]));
+                    });
+                }
+            );
+        }
+        
         [Fact]
         public async Task SendHttpPostRequest_WithContinueTimeoutAndBody_BodyIsDelayed()
         {
+            TaskCompletionSource tcs = new TaskCompletionSource();
             await LoopbackServer.CreateClientAndServerAsync(
                 async (uri) =>
                 {
                     HttpWebRequest request = WebRequest.CreateHttp(uri);
                     request.Method = "POST";
                     request.ServicePoint.Expect100Continue = true;
                     request.ContinueTimeout = 30000;
-                    Stream requestStream = await request.GetRequestStreamAsync();
-                    requestStream.Write("aaaa\r\n\r\n"u8);
+                    request.ContentLength = 8;
+                    using (Stream requestStream = await request.GetRequestStreamAsync())
+                    {
+                        requestStream.Write("aaaa\r\n\r\n"u8);
+                    }
```

Comment:
Are we intentionally disposing the stream before getting the response? Can't this result in a race?